File: sol/3147/3147.py
class Solution:
    def maximumEnergy(self, energy: List[int], k: int) -> int:
        n = len(energy)
        maxval = max(energy[n - k:])
        for i in range(n - k - 1, -1, -1):
            energy[i] += energy[i + k]
            maxval = max(maxval, energy[i])
        return maxval

File: sol/3147/3147.cpp
struct Solution {
    int maximumEnergy(vector<int>& energy, int k) {
        int n = energy.size();
        int maxval = energy[n - k];
        for (int i = n - k + 1; i < n; ++i){
            maxval = max(maxval, energy[i]);
        }
        for (int i = n - k - 1; i >= 0; --i){
            energy[i] += energy[i + k];
            maxval = max(maxval, energy[i]);
        }
        return maxval;
    }
};
